test/test.js

```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-inner-declarations */
```

Comment:
What is an inner-declaration?

Reply:
It's a function declaration that isn't at the top function-level scope (i.e. it's in a try block, an if statement, etc.).  AFAICT, ESlint complains because hoisting the function means it is not a ReferenceError to use it outside the block, but it is undefined until the block evaluates.

For generator functions that rely on block-scoped closed-over variables, this is a puzzle.  There is no arrow-function equivalent, and making them anonymous triggers another ESlint warning.

In the next commit, I lift the inner generator functions to the enclosing function scope, along with let-declarations of variables the generators close over.  Talk about working around the linter!